packages/cli/src/spec/mock/MockPromptForStrings.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import {Prompt, ValidateFunction} from '../../lib/Prompt';
+
+/**
+ * A class which used for testing and which mocks user's input.
+ */
+export class MockPromptForStrings implements Prompt {
+  private responses: string[] = [];
+  private lastMessageIndex = -1;
```

Comment:
Can you not replace `lastMessageIndex` everywhere with `responses.length`?

Reply:
I can, I had problems with push and pop so I did what I did, but this is a good solution